sunpy/map/mapbase.py

```diff
@@ -1291,108 +1347,56 @@ def superpixel(self, dimensions, offset=(0, 0)*u.pixel, func=np.sum):
         -------
-        lines: list
-            A list of `matplotlib.lines.Line2D` objects that have been plotted.
-
-        Notes
-        -----
-        keyword arguments are passed onto matplotlib.pyplot.plot
+        overlay: `~astropy.visualization.wcsaxes.coordinates_map.CoordinatesMap`
+            The wcsaxes coordinate overlay instance.
         """
 
         if not axes:
             axes = wcsaxes_compat.gca_wcs(self.wcs)
-
-        lines = []
-
-        # Do not automatically rescale axes when plotting the overlay
-        axes.set_autoscale_on(False)
-
-        transform = wcsaxes_compat.get_world_transform(axes)
-
-        XX, YY = np.meshgrid(np.arange(self.data.shape[0]),
-                             np.arange(self.data.shape[1]))
-        x, y = self.pixel_to_data(XX*u.pix, YY*u.pix)
-        dsun = self.dsun
-
-        b0 = self.heliographic_latitude.to(u.deg).value
-        l0 = self.heliographic_longitude.to(u.deg).value
-        units = self.spatial_units
-
-        # Prep the plot kwargs
-        plot_kw = {'color': 'white',
-                   'linestyle': 'dotted',
-                   'zorder': 100,
-                   'transform': transform}
-        plot_kw.update(kwargs)
-
-        hg_longitude_deg = np.linspace(-180, 180, num=361) + l0
-        hg_latitude_deg = np.arange(-90, 90, grid_spacing.to(u.deg).value)
-
-        # draw the latitude lines
-        for lat in hg_latitude_deg:
-            x, y = wcs.convert_hg_hpc(hg_longitude_deg, lat * np.ones(361),
-                                      b0_deg=b0, l0_deg=l0, dsun_meters=dsun,
-                                      angle_units=units.x, occultation=True)
-            valid = np.logical_and(np.isfinite(x), np.isfinite(y))
-            x = x[valid]
-            y = y[valid]
-            if wcsaxes_compat.is_wcsaxes(axes):
-                x = (x*u.arcsec).to(u.deg).value
-                y = (y*u.arcsec).to(u.deg).value
-            lines += axes.plot(x, y, **plot_kw)
-
-        hg_longitude_deg = np.arange(-180, 180, grid_spacing.to(u.deg).value) + l0
-        hg_latitude_deg = np.linspace(-90, 90, num=181)
-
-        # draw the longitude lines
-        for lon in hg_longitude_deg:
-            x, y = wcs.convert_hg_hpc(lon * np.ones(181), hg_latitude_deg,
-                                      b0_deg=b0, l0_deg=l0, dsun_meters=dsun,
-                                      angle_units=units[0], occultation=True)
-            valid = np.logical_and(np.isfinite(x), np.isfinite(y))
-            x = x[valid]
-            y = y[valid]
-            if wcsaxes_compat.is_wcsaxes(axes):
-                x = (x*u.arcsec).to(u.deg).value
-                y = (y*u.arcsec).to(u.deg).value
-            lines += axes.plot(x, y, **plot_kw)
-
-        # Turn autoscaling back on.
-        axes.set_autoscale_on(True)
-        return lines
+        if not wcsaxes_compat.is_wcsaxes(axes):
+            raise TypeError("Overlay grids can only be plotted on WCSAxes plots.")
```

Comment:
when would be possible that this happens? in such case, how the person interested could get the grid?